cdptools/databases/document_store_database.py

```diff
@@ -0,0 +1,840 @@
+            for that table.
+        """
+        # Cast to dataframe
+        formatted = pd.DataFrame(rows)
+
+        # Optionally set the index based on the table name
+        if table:
+            formatted = formatted.set_index(f"{table}_id")
+
+        return formatted
+
+    @abstractmethod
+    def get_or_upload_body(
+        self,
+        name: str,
+        description: Optional[str],
+        start_date: datetime,
+        end_date: Optional[datetime],
+        is_active: bool,
+        chair_person_id: str,
+        external_source_id: Optional[Any],
+    ) -> Dict[str, Any]:
+        """
+        Get or upload a body.
+
+        Parameters
+        ----------
+        name: str
+            The name of the body (council, subcommittee, etc).
+        description: Optional[str]:
+            An optional description of the body.
+        start_date: datetime
+            A datetime representing the chronological start date of a body.
+        end_date: Optional[datetime]
+            A datetime representing the end date (if not active) of a body.
+        is_active: bool
+            A bool describing whether a body is active or not.
+        chair_person_id: str
+            The id of the body chair's person row.
+        external_source_id: Optional[Any]
+            An id in the external source this data comes from.
+
+        Returns
+        -------
+        details: Dict[str, Any]
+            A dictionary containing the data that was either uploaded or found.
+        """
+        return {}
+
+    @abstractmethod
+    def get_or_upload_minutes_item(
+        self,
+        name: str,
+        description: Optional[str],
+        matter: Dict[str, str],
+        external_source_id: Optional[Any],
+    ) -> Dict[str, Any]:
+        """
+        Get or upload a minutes item. In Legistar this is commonly referred to 
+        as an event item.
+
+        Parameters
+        ---------
+        name: str
+            A name for the minutes item. Ex: "Appointment of Rene J. Peters, Jr."
+        description: Optional[str]
+            An optional description of the minutes_item.
+        matter: Optional[str]
+            A dictionary that contains information of the matter for the minutes item.
+        external_source_id: Optional[Any]
+            An id in the external source this data comes from.
+
+        Returns
+        -------
+        details: Dict[str, Any]
+            A dictionary containing the data that was either uploaded or found.
+        """
+        return {}
+
+    @abstractmethod
+    def get_or_upload_event(
+        self,
+        body: Dict[str, str],
+        event_datetime: datetime,
+        thumbnail_static_file: Dict[str, str],
+        thumbnail_hover_file: Optional[str],
+        video_uri: Optional[str],
+        keywords: Dict[str, str],
+        matters: Dict[str, str],
+        minutes_items: Dict[str, str],
+        people: Dict[str, str],
```

Comment:
I think  `keywords`, `matters`, `minutes_items`, and `people` should be typed List of Dict?